modules/src/core/ics04_channel/packet.rs

```diff
@@ -110,6 +112,32 @@ pub struct Packet {
     pub timeout_timestamp: Timestamp,
 }
 
+struct PacketData<'a>(&'a [u8]);
+
+impl<'a> ::core::fmt::Debug for PacketData<'a> {
+    fn fmt(&self, formatter: &mut ::core::fmt::Formatter<'_>) -> Result<(), ::core::fmt::Error> {
+        write!(formatter, "{:?}", self.0)
+    }
+}
+
+impl ::core::fmt::Debug for Packet {
+    fn fmt(&self, formatter: &mut ::core::fmt::Formatter<'_>) -> Result<(), ::core::fmt::Error> {
+        let data_wrapper = PacketData(&self.data);
```

Comment:
Thanks, @plafer! Nicely done! 👌 
Might be helpful to destructure the packet here so that the build fails when we forget to update the debug impl after we modify the `Packet` struct. And maybe add a comment here to remind us to update the code below when that happens.
```suggestion
        let Packet {
            sequence,
            source_port,
            source_channel,
            destination_port,
            destination_channel,
            data,
            timeout_height,
            timeout_timestamp,
        } = self;
        let data_wrapper = PacketData(data);
```